llvm/lib/CodeGen/InterleavedAccessPass.cpp

```diff
@@ -545,7 +545,8 @@ bool InterleavedAccessImpl::runOnFunction(Function &F) {
   }
 
   for (auto *I : DeadInsts)
-    I->eraseFromParent();
+    if (I->getParent())
```

Comment:
It looks like you're trying to fix a scenario where the same Instruction is included twice in `DeadInsts`, right? I think the problem here is that `I` may have been deleted and so you're dereferencing unreliable contents in memory or we could seg fault. I think you probably need to fix this a different way either by:

1. Keeping track of which instructions you've deleted, i.e. by maintaining a map of visited instructions. Or,
2. Removing duplicate dead instructions from the list before iterating over them.